ecs-agent/netlib/platform/cniconf_windows.go

```diff
@@ -0,0 +1,144 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+//go:build windows
+// +build windows
+
+package platform
+
+import (
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/aws/amazon-ecs-agent/ecs-agent/netlib/model/ecscni"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/netlib/model/networkinterface"
+
+	"github.com/containernetworking/cni/pkg/types"
+)
+
+const (
+	VPCENIPluginName = "vpc-eni.exe"
+	// taskNetworkNamePrefix is the prefix of the HNS network used for task ENI.
+	taskNetworkNamePrefix = "task"
+	// fargateBridgeNetworkName is the name of the HNS network which is used for task IAM roles endpoint.
+	fargateBridgeNetworkName = "fargate-bridge"
+
+	// ENISetupTimeout is the maximum duration that ENI manager waits before aborting ENI setup.
+	// The reasoning for this timeout duration is explained below with nsSetupTimeoutDuration.
+	ENISetupTimeout = 3 * time.Minute
+	// nsCleanupTimeoutDuration is irrelevant for warmpool instances since they are recycled after each
+	// task and therefore, DEL operation by CNI plugin is executed without any config.
+	// This value has been made similar to the one present for Linux.
+	nsCleanupTimeoutDuration = 2 * time.Second
+	// Timeout duration for each network setup and cleanup operation before it is cancelled.
+	// The creation of networking stack on Windows is dependent upon HNS, which is an inbox network orchestration
+	// service provided by Windows. This can lead to failures and/or increased latency in networking setup.
+	// The ideal network setup time would be less than 30 seconds.These network setup timeout values are selected
+	// after many trails which allow the tasks to be launched comfortably. These values are similar to the ones
+	// used by ECS agent during CNI plugin invocation.
+	nsSetupTimeoutDuration = 45 * time.Second
+
+	// Values for creating backoff while retrying network setup.
+	// As mentioned above, the networking stack creation is dependent upon HNS. During our POCs, it was found that
```

Comment:
nit: could you add the full name of HNS here? I would be hard for someone who doesn't have any context to search for this acronym to learn more about it.